### PR TITLE
update to how offers are retrieved

### PR DESCRIPTION
**Pull request by sbenedicadb**
rather than using bundle and org to retrieve offers, the new format requires only the bundle (and a new json format).  this change updates to the new format and adjusts tests accordingly

### CODE REVIEW

File: AEPMessaging/Tests/UnitTests/MessagingTests.swift [hunk 1]
```diff
@@ -146,11 +145,10 @@ class MessagingTests: XCTestCase {
         XCTAssertNotNil(mockCache.setParamEntry)
     }
 
-    func testHandleOfferNotificationMissmatchedActivity() throws {
+    func testHandleOfferNotificationMissmatchedBundle() throws {
```
Comment (rymorale) [inline AEPMessaging/Tests/UnitTests/MessagingTests.swift:148]: nit: typo in `Mismatched`

File: AEPMessaging/Sources/Messaging.swift [hunk 2]
```diff
@@ -222,33 +238,41 @@ public class Messaging: NSObject, Extension {
...
+    private func getOffersMessageConfig() -> (String?, String?) {
+        if let path = Bundle.main.path(forResource: "Info", ofType: "plist"), let plistDictionary = NSDictionary(contentsOfFile: path) {
+            guard let activity = plistDictionary.value(forKey: MessagingConstants.IAM.Plist.ACTIVITY_ID) as? String,
+                  let placement = plistDictionary.value(forKey: MessagingConstants.IAM.Plist.PLACEMENT_ID) as? String else {
+                      return (nil, Bundle.main.bundleIdentifier)
+                  }
+            return (activity, placement)
+        }
+        
+        return (nil, Bundle.main.bundleIdentifier)
```
Comment (rymorale) [inline AEPMessaging/Sources/Messaging.swift:275]: nit: would this be a bit cleaner if the logic was:
```swift
if let path = Bundle.main.path(forResource: "Info", ofType: "plist"), let plistDictionary = NSDictionary(contentsOfFile: path) {
            if let activity = plistDictionary.value(forKey: MessagingConstants.IAM.Plist.ACTIVITY_ID) as? String,
                  let placement = plistDictionary.value(forKey: MessagingConstants.IAM.Plist.PLACEMENT_ID) as? String {
                    return (activity, placement)
                  }
        }
    return (nil, Bundle.main.bundleIdentifier)
```

**Review by rymorale — APPROVED** [review]
just a couple of minor issues (misspelled test name and a nit with the getOffersMessageConfig logic). otherwise looks good to me.

File: AEPMessaging/Sources/Event+Messaging.swift [hunk 3]
```diff
@@ -226,6 +227,22 @@ extension Event {
...
+    var offerDecisionScope: String? {
+        guard let payload = payload, !payload.isEmpty else {
+            return nil
+        }
+        
+        guard let b64EncodedScope = payload.first?[MessagingConstants.Event.Data.Key.Optimize.SCOPE] as? String else {
+            return nil
+        }
+        
+        guard let scopeData = Data(base64Encoded: b64EncodedScope), let scopeDictionary = try? JSONSerialization.jsonObject(with: scopeData, options: .mutableContainers) as? [String: Any] else {
```
Comment (PravinPK) [inline AEPMessaging/Sources/Event+Messaging.swift:240]: Any specific requirement to keep the JSONSerialization.ReadingOptions as `mutableContainers ` ?

File: AEPMessaging/Sources/Message+FullscreenMessageDelegate.swift [hunk 4]
```diff
@@ -42,7 +42,7 @@ extension Message: FullscreenMessageDelegate {
     ///   - message: the message attempting to load a URL
     ///   - url: the URL attempting to be loaded
     /// - Returns: false if the message's webview will handle the loading of the URL
-    public func overrideUrlLoad(message: FullscreenMessage, url: String?) -> Bool {
+    public func overrideUrlLoad(message fullscreenMessage: FullscreenMessage, url: String?) -> Bool {
```
Comment (PravinPK) [inline AEPMessaging/Sources/Message+FullscreenMessageDelegate.swift:45]: change the docs above according to the renaming 
'message' -> `fullscreenMessage`

File: AEPMessaging/Sources/Message+FullscreenMessageDelegate.swift [hunk 5]
```diff
@@ -69,7 +69,7 @@ extension Message: FullscreenMessageDelegate {
 
             // dismiss if requested
             if url.host == MessagingConstants.IAM.HTML.DISMISS {
-                message?.dismiss(suppressAutoTrack: true)
+                onDismiss(message: fullscreenMessage)
```
Comment (sbenedicadb) [inline AEPMessaging/Sources/Message+FullscreenMessageDelegate.swift:72]: i have to revert this change, it's causing double tracking
